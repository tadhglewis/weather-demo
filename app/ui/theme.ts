import theme from "styled-theming";

export const primary = theme("theme", {
  light: "#49b0cf",
  dark: "#354d71",
});

export const background = theme("theme", {
  light: "white",
  dark: "#1e1e1e",
});

export const textPrimary = theme("theme", {
  light: "white",
  dark: "white",
});

export const textSecondary = theme("theme", {
  light: "#e3e3e3",
  dark: "#e3e3e3",
});
