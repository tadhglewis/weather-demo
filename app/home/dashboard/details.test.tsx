import React from "react";
import { render } from "@testing-library/react-native";
import Details from "./details";
import { ThemeProvider } from "styled-components";
import "jest-styled-components";
import theme from "styled-theming";

// mocking and always return light theme due to a bug where ThemeProvider isn't passing in the theme to styled-theming in tests
jest.mock("styled-theming", () => (name: string, values: theme.ThemeMap) =>
  values.light
);

describe("details", () => {
  it("should render details", () => {
    const tree = render(
      // <ThemeProvider theme={{ theme: "light" }}>
      <Details
        items={[
          { label: "test", value: "123" },
          { label: "hello", value: "hi" },
        ]}
      />
      // </ThemeProvider>
    );

    expect(tree.getByText("test")).toBeDefined();
    expect(tree.getByText("hello")).toBeDefined();

    expect(tree).toMatchSnapshot();
  });
});
