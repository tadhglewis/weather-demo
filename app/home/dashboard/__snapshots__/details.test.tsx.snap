// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`details should render details 1`] = `
<View
  style={
    Array [
      Object {
        "marginTop": 32,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "width": "30%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      test
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#e3e3e3",
            "fontSize": 16,
          },
        ]
      }
    >
      123
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "width": "30%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      hello
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#e3e3e3",
            "fontSize": 16,
          },
        ]
      }
    >
      hi
    </Text>
  </View>
</View>
`;
